evennia/utils/opclasses.py

```diff
@@ -0,0 +1,278 @@
+import datetime as _dt
+from evennia import logger as _log
+from evennia.utils.ansi import ANSIString as _ANSI
+from evennia.utils.validfuncs import _TZ_DICT
+from evennia.utils.containers import VALID_CONTAINER as _VAL
+
+
+class BaseOption(object):
+    """
+    Abstract Class to deal with encapsulating individual Options. An Option has a name/key, a description
+    to display in relevant commands and menus, and a default value. It saves to the owner's Attributes using
+    its Handler's save category.
+
+    Designed to be extremely overloadable as some options can be cantankerous.
+
+    Properties:
+        expect_type (str): What users will see this as asking for. Example: Color or email.
+        valid: Shortcut to the loaded VALID_HANDLER.
+        valid_type (str): The key of the Validator this uses.
+    """
+    validator_key = ''
+
+    def __str__(self):
+        return self.key
+
+    def __init__(self, handler, key, description, default, save_data=None):
+        """
+
+        Args:
+            handler (OptionHandler): The OptionHandler that 'owns' this Option.
+            key (str): The name this will be used for storage in a dictionary. Must be unique per
+                OptionHandler.
+            description (str): What this Option's text will show in commands and menus.
+            default: A default value for this Option.
+            save_data: Whatever was saved to Attributes. This differs by Option.
+        """
+        self.handler = handler
+        self.key = key
+        self.default_value = default
+        self.description = description
+        self.save_data = save_data
+
+        # Value Storage contains None until the Option is loaded.
+        self.value_storage = None
+
+        # And it's not loaded until it's called upon to spit out its contents.
+        self.loaded = False
+
+    def display(self, **kwargs):
+        """
+        Renders the Option's value as something pretty to look at.
+
+        Returns:
+            How the stored value should be projected to users. a raw timedelta is pretty ugly, y'know?
+        """
+        return self.value
+
+    def _load(self):
```

Comment:
In light of discussing using `options` also for, what today's `@options` command handles (client flags), then my request to make this private no longer applies - this and `save` would have to be overloaded and should then not be private - sorry. :)